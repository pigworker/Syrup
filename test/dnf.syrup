not(<Bit>) -> <Bit>
not(X) = nand(X,X)

and(<Bit>, <Bit>) -> <Bit>
and(X,Y) = not(nand(X,Y))

or(<Bit>, <Bit>) -> <Bit>
or(X,Y) = nand(Z, S) where
  Z = not(X)
  S = not(Y)

one() -> <Bit>
one() = !0

























----------------------------------------------

complex(<Bit>, <Bit>, <Bit>) -> <Bit>
complex(X,Y,Z) = !(X|Y&Z) & !(!Z&Y&X)

dnf complex