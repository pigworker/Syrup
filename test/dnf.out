Circuit not is defined.

Circuit and is defined.

Circuit or is defined.

Circuit complex is defined.

Disjunctive Normal Form of complex:
complex(<Bit>, <Bit>, <Bit>) -> <Bit>
complex(X, Y, Z) = !X & !Y | !X & !Z
